src/app/components/compact-list/compact-list.component.ts

```diff
@@ -0,0 +1,184 @@
+import {Component, Input, Output, ElementRef, Renderer, ViewChild} from "@angular/core";
+import {FormGroup, FormControl, Validators} from "@angular/forms";
+import {ReplaySubject} from "rxjs";
+import {ComponentBase} from "../common/component-base";
+import {ContenteditableDirective} from "./directives/contenteditable.directive";
+import {TagModel} from "./tag.model";
+
+require("./compact-list.component.scss");
+
+export interface ValidationResponse {
+    isValid: boolean,
+    errorMessage: string
+}
+
+@Component({
+    selector: "compact-list",
+    directives: [
+        ContenteditableDirective
+    ],
+    template: `
+       <form [formGroup]="formGroup" class="compact-list-form">
+            <div class="compact-list-wrapper" (click)="onListWrapperClick()">
+                
+                <div class="input-tag-list">
+                    <span *ngFor="let tag of tagList; let i = index; trackBy:trackByIndex"
```

Comment:
I was under the impression that `trackBy` works with indices by default and is usually not needed unless the list has duplicates. [Or for performance reasons, as suggested by Angular University.](http://blog.angular-university.io/angular-2-ngfor/)